src/main/java/ResourceDiscovery/AssetObjects/AssetObject.java

```diff
@@ -182,32 +203,61 @@ public String getStatus() {
         return this.status;
     }
 
+    /**
+     * Get the AssetType of this object.
+     * @return A enum of AssetTypes representing the specific type of this Asset Object.
+     */
+    public AssetTypes getAssetTypeEnum() {
+        return this.assetTypeEnum;
+    }
+
     /*
      * This function receives a string representing a url and extracts the string after the last '/'
      * char in the url.
      * If the provided string does not match that pattern the original string is returned.
      */
-    protected static String getLastSeg(String url) {
-        Matcher matcher = LAST_SEGMENT_PATTERN.matcher(url);
+    protected static String getLastSeg(Object url) {
+        String urlToReturn = (String) url;
+        Matcher matcher = LAST_SEGMENT_PATTERN.matcher(urlToReturn);
         if (matcher.find()) {
             return matcher.replaceAll("");
         }
-        return url;
+        return urlToReturn;
+    }
+
+    /*
+     * This function receives a list of strings representing urls and returns a list of strings
+     * with the last segment for each url.
+     */
+    protected static List<String> convertListToLastSegList(Object urlsObject) {
```

Comment:
I wonder if this can be rephrased as a ".stream().map(url ->" etc, possibly being so short as to allow to removing this function entirely

Reply:
I changed this function according to your comment, however I do think it is still worthwhile having it as a function because it is used in several places and also for casting validation. Sounds okay?

Reply:
LGTM